File: sorting/05_insertionSort.py
def insertionSort(A):

    valueIndex = 0
    value = 0

    for i in range(1, len(A)):
        valueIndex = i
        value = A[i]

        while valueIndex > 0 and A[valueIndex - 1] > value:
          A[valueIndex] = A[valueIndex - 1];
          valueIndex = valueIndex - 1;

        A[valueIndex] = value

    return A

File: js/sorting/05_insertionSort.js
const insertionSort = (A = []) => {

  let valueIndex;
  let value;

  for (let j = 1; j < A.length; j++) {
    value = A[j];
    valueIndex = j;

    while (valueIndex > 0 && A[valueIndex - 1] > value) {
      A[valueIndex] = A[valueIndex - 1];
      valueIndex = valueIndex - 1;
    }

    A[valueIndex] = value;
  }

  return A;
}

console.log(insertionSort([17,5,12,2,20,3,7,1,0]));
